pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +384,94 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+# Det Diag Rewrites
+@register_canonicalize
+@register_stabilize
+@node_rewriter([det])
+def rewrite_det_diag_from_eye_mul(fgraph, node):
+    """
+     This rewrite takes advantage of the fact that for a diagonal matrix, the determinant value is the product of its diagonal elements.
+
+    The presence of a diagonal matrix is detected by inspecting the graph. This rewrite can identify diagonal matrices that arise as the result of elementwise multiplication with an identity matrix. Specialized computation is used to make this rewrite as efficient as possible, depending on whether the multiplication was with a scalar, vector or a matrix.
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        Function graph being optimized
+    node: Apply
+        Node of the function graph to be optimized
+
+    Returns
+    -------
+    list of Variable, optional
+        List of optimized variables, or None if no optimization was performed
+    """
+    # Find if we have Blockwise Op
+    if not (isinstance(node.op, Blockwise) and isinstance(node.op.core_op, Det)):
+        return None
+
+    node_input_op = node.inputs[0]
+    # Check if the op is Elemwise and mul
+    if not (
+        node_input_op.owner is not None
+        and isinstance(node_input_op.owner.op, Elemwise)
+        and isinstance(node_input_op.owner.op.scalar_op, Mul)
+    ):
+        return None
+
+    # Find whether any of the inputs to mul is Eye
+    inputs_to_mul = node_input_op.owner.inputs
+    eye_check = False
+    eye_input = []
+    for mul_input in inputs_to_mul:
+        if isinstance(mul_input.owner.op, Eye):
+            eye_check = True
+            eye_input.append(mul_input)
+            break  # This is so that we only find the first one? Can be changed
+    if not (eye_check):
+        return None
```

Comment:
```suggestion
    eye_input = []
    for mul_input in inputs_to_mul:
        if isinstance(mul_input.owner.op, Eye):
            eye_input.append(mul_input)
            break  # This is so that we only find the first one? Can be changed
    else:  #no-break
        return None
```